_posts/2019-10-01-intro-to-hmc/hmc-intro.Rmd

```diff
@@ -0,0 +1,283 @@
+author:
+  - name: Sigrid Keydana
+    affiliation: RStudio
+    affiliation_url: https://www.rstudio.com/
+slug: keydana2019hmc
+date: 10-01-2019
+categories:
+  - Probability and statistics
+  - Concepts
+  - Introductions
+bibliography: bibliography.bib
+output:
+  distill::distill_article:
+    self_contained: false
+preview: images/mb.png
+---
+
+```{r setup, include=FALSE}
+knitr::opts_chunk$set(echo = TRUE, eval = FALSE)
+```
+
+Why a _very (meaning: VERY!) first conceptual introduction to Hamiltonian Monte Carlo_ (HMC) on this blog? 
+
+Well, in our endeavor to feature the various capabilities of TensorFlow Probability (TFP) / [tfprobability](https://rstudio.github.io/tfprobability/), we started showing examples of how to fit hierarchical models, using one of TFP's joint distribution classes ^[[tfd_joint_distribution_sequential](https://rstudio.github.io/tfprobability/reference/tfd_joint_distribution_sequential.html)] and HMC. The technical aspects being complex enough in themselves, we never provided an introduction to the "math side of things". For those of our readers who are familiar with the technique (Stan users, for example), we hope this was acceptable; but how about those who are following the blog -- the _TensorFlow for R_ blog, alright -- due to their interest in _deep learning_? They may have felt left behind or -- put more prosaically -- just have stopped reading instantaneously.
+
+This post is directed towards the second group of readers, aiming to make up for the imposition.
+If after reading, you still feel confused, it isn't your fault and _maybe_, not even the fault of this text. Based on an inofficial, minimal-sample-size survey, this topic is a _lot_ more confusing than deep learning :-). As to the content, we'll try to keep it as digestable as possible, while at the same time striving to mention all of the famous incantations: Hamiltonian Monte Carlo, _leapfrog_ steps, NUTS.
+
+
+## So what is it for?
+
+Sampling, or _Monte Carlo_, techniques in general are used when we want to produce samples from, or statistically describe, a distribution we don't have a closed-form formulation of. Sometimes, we might really be interested in the samples; sometimes we just want them so we can compute, for example, the mean and variance of the distribution.
+
+What distribution? In the type of applications we're talking about, we have a _model_, a joint distribution, supposed to describe some reality. Starting from the most basic scenario, it might look like this:
```

Comment:
```suggestion
What distribution? In the type of applications we're talking about, we have a _model_, a joint distribution, which is supposed to describe some reality. Starting from the most basic scenario, it might look like this:
```